### PR TITLE
Checker service refactor (v3-1871)

### PR DESCRIPTION
**Pull request by aligeti**
What: 
Refactor checker service
https://storjlabs.atlassian.net/browse/V3-1871

Why:
Looks cleaner and consistent

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/datarepair/checker/checker.go [hunk 1]
```diff
@@ -247,6 +246,20 @@ func (checker *Checker) updateSegmentStatus(ctx context.Context, pointer *pb.Poi
 	return nil
 }
 
+// IrreparableSegmentCron runs checker's Irreparable process service
+func (checker *Checker) IrreparableSegmentCron(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	checker.logger.Info("checker's Irreparable process cron is starting up")
+
+	return checker.IrreparableLoop.Run(ctx, func(ctx context.Context) error {
+		err := checker.IrreparableProcess(ctx)
```
Comment (wthorp) [inline pkg/datarepair/checker/checker.go]: just return the err...
`return checker.IrreparableProcess(ctx)`

Comment (wthorp) [reply pkg/datarepair/checker/checker.go]: or a wrapped  version of the err

File: pkg/datarepair/checker/checker.go [hunk 2]
```diff
@@ -75,34 +77,17 @@ func NewChecker(metainfo *metainfo.Service, repairQueue queue.RepairQueue, overl
...
-		c <- checker.IrreparableLoop.Run(ctx, func(ctx context.Context) error {
-			err := checker.IrreparableProcess(ctx)
-			if err != nil {
-				checker.logger.Error("error with irreparable segments identification", zap.Error(err))
-			}
-			return nil
-		})
-	}()
+	group.Go(func() error {
+		return errs2.IgnoreCanceled(checker.IrreparableSegmentCron(ctx))
```
Comment (wthorp) [inline pkg/datarepair/checker/checker.go]: you could probably use an inline / anonymous function here (similar to how you had it before) and increase readability.

Comment (aligeti) [reply pkg/datarepair/checker/checker.go]: updated.

File: pkg/datarepair/checker/checker.go (hunk 1 above)
Comment (aligeti) [reply pkg/datarepair/checker/checker.go]: used inline

**Pull request merged**